From bf3c821903a330ccb054a561dd9048d96b80e471 Mon Sep 17 00:00:00 2001
From: Jasper Korten <jja2000@gmail.com>
Date: Wed, 5 May 2021 17:43:18 +0500
Subject: [PATCH 190/210] FIXME: arm64: dts: qcom: msm8916-samsung-gt5-common:
 Add audio jack detection

gt5 devices use jack detection IC. Add it.

Signed-off-by: Jasper Korten <jja2000@gmail.com>
Co-developed-by: Nikita Travkin <nikita@trvn.ru>
Signed-off-by: Nikita Travkin <nikita@trvn.ru>
---
 .../dts/qcom/msm8916-samsung-gt5-common.dtsi   | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi b/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi
index 5287ff21a15..1aab125b473 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi
@@ -210,6 +210,16 @@ &usb_hs_phy {
 	extcon = <&pm8916_usbin>;
 };
 
+&wcd_codec {
+	jack-gpios = <&msmgpio 110 GPIO_ACTIVE_LOW>;
+	qcom,micbias-lvl = <2800>;
+	qcom,mbhc-vtreshold-low = <75 150 237 450 500>;
+	qcom,mbhc-vtreshold-high = <75 150 237 450 500>;
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&jack_default>;
+};
+
 &smd_rpm_regulators {
 	vdd_l1_l2_l3-supply = <&pm8916_s3>;
 	vdd_l4_l5_l6-supply = <&pm8916_s4>;
@@ -345,4 +355,12 @@ gpio_hall_sensor_default: gpio-hall-sensor-default {
 		drive-strength = <2>;
 		bias-disable;
 	};
+
+	jack_default: jack-default {
+		pins = "gpio110";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
 };
-- 
2.34.1

